app/src/server/tasks/generateTestSetEntry.task.ts

```diff
@@ -19,166 +19,194 @@ import {
+        } else if (fineTune) {
+          completion = await getCompletion2(fineTune, input);
+        } else {
+          await prisma.fineTuneTestingEntry.update({
+            where: { modelId_datasetEntryId: { modelId, datasetEntryId } },
+            data: {
+              errorMessage: "The model is not set up for inference",
+            },
+          });
+          return;
+        }
 
-    try {
-      if (isComparisonModel(modelId)) {
-        completion = await getOpenaiCompletion(rawDatasetEntry.dataset.projectId, input);
-      } else if (fineTune) {
-        completion = await getCompletion2(fineTune, input);
-      } else {
+        const choice = completion.choices[0];
+        if (!choice) throw new Error("No completion returned");
+        const completionMessage = choice.message;
+
+        if (fineTune) {
+          const inputTokens = completion.usage?.prompt_tokens ?? 0;
+          const outputTokens = completion.usage?.completion_tokens ?? 0;
+          await prisma.usageLog.create({
+            data: {
+              fineTuneId: fineTune.id,
+              type: UsageType.TESTING,
+              inputTokens,
+              outputTokens,
+              cost: calculateFineTuneUsageCost({
+                inputTokens,
+                outputTokens,
+                baseModel: fineTune.baseModel,
+              }),
+            },
+          });
+        }
+
+        const updatedFineTuneTestingEntry = await prisma.fineTuneTestingEntry.update({
+          where: { modelId_datasetEntryId: { modelId, datasetEntryId } },
+          data: {
+            cacheKey,
+            output: completionMessage as unknown as Prisma.InputJsonValue,
+            finishReason: choice.finish_reason,
+            prunedInputTokens: completion.usage?.prompt_tokens,
+            outputTokens: completion.usage?.completion_tokens,
+            errorMessage: null,
+          },
+        });
+
+        await triggerEvals(datasetEntry, updatedFineTuneTestingEntry);
+      } catch (e: unknown) {
+        const typedError = e as { message?: string; error?: { message: string } };
         await prisma.fineTuneTestingEntry.update({
           where: { modelId_datasetEntryId: { modelId, datasetEntryId } },
           data: {
-            errorMessage: "The model is not set up for inference",
+            errorMessage:
+              typedError.message ?? typedError.error?.message ?? "Error retrieving completion",
           },
         });
-        return;
+        throw e;
       }
-
-      const choice = completion.choices[0];
-      if (!choice) throw new Error("No completion returned");
-      const completionMessage = choice.message;
-
-      if (fineTune) {
-        const inputTokens = completion.usage?.prompt_tokens ?? 0;
-        const outputTokens = completion.usage?.completion_tokens ?? 0;
-        await prisma.usageLog.create({
+    } catch (e) {
+      console.error("error in generateTestSetEntry", e);
```

Comment:
We should probably report this to Sentry?

Reply:
If we do that, we'll end up getting a bunch of errors due to rate limiting. We could insert a check to avoid reporting errors caused by rate limiting, but that should go in a separate PR.